src/main/java/com/checkmarx/jenkins/CxScanBuilder.java

```diff
@@ -1318,7 +1318,9 @@ private CxScanConfig resolveConfiguration(Run<?, ?> run, DescriptorImpl descript
         } else {
             ret.setProxy(false);
         }
-        teamPath = getTeamNameFromId(cxConnectionDetails, descriptor, groupId);
+        if (teamPath == null) {
+            teamPath = getTeamNameFromId(cxConnectionDetails, descriptor, groupId);
```

Comment:
Will this actually fix the issue?

The title of the PR says "when no `groupId` is provided, however here it is instead testing `teamPath == null`, and contuing to use `groupId` as a paramter.

Reply:
Nevermind, it looks good to me, based on your description

> This fix prevents the CxCommonClient from throwing a NullPointerException when the `groupId` is null if the `teamPath` parameter has some value.

So the idea is to skip this step if the `teamPath` is already set with something.